### PR TITLE
xlsxio: add new recipe

### PR DESCRIPTION
**Pull request by hiiizxf**
Specify library name and version:  xlsxio/*

lib xlsxio is usefull to reading values from and writing values to .xlsx files. 
I would like to see it in conan center.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/xlsxio/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        "with_minizip_ng": [True, False],
+        "with_wide": [True, False],
+        "build_documentation": [True, False],
+    }
+    default_options = {
+        "fPIC": True,
+        "shared": False,
+        "build_pc_files": True,
+        "build_tools": True,
+        "build_examples": True,
```
[the remark this answers is not in the document]
Comment (danimtb) [reply recipes/xlsxio/all/conanfile.py:37]: Maybe even delete this option in the recipe. CMake options != recipe options. If the options do not make sense for the user input view, then it is fine to nop add it here

File: recipes/xlsxio/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        "with_libzip": [True, False],
+        "with_minizip_ng": [True, False],
+        "with_wide": [True, False],
+        "build_documentation": [True, False],
+    }
+    default_options = {
+        "fPIC": True,
+        "shared": False,
+        "build_pc_files": True,
+        "build_tools": True,
```
Comment (danimtb) [inline recipes/xlsxio/all/conanfile.py:36]: Do you know why these options are enabled? Does it make sense when using Conan  (as conan generates pc files if you chose the generator for so)?

Are build tools required here as well? What do they do?

File: recipes/xlsxio/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        "build_examples": True,
+        "with_libzip": False,
+        "with_minizip_ng": False,
+        "with_wide": False,
+        "build_documentation": False,
```
Comment (danimtb) [inline recipes/xlsxio/all/conanfile.py:41]: same here with the documentation option. I think it is not needed here

**Review by danimtb — COMMENTED** [review]
Great job in the recipe! I see some issues with the options though. Please take a look at the comments. I think we can simplify them 😄

File: recipes/xlsxio/all/conanfile.py (hunk 2 above)
Comment (hiiizxf) [reply recipes/xlsxio/all/conanfile.py:36]: 1.The pc file is generated from template.pc.in when build with cmake, when using Conan it do not make sense, i will remove it.
2.Here "build_tools" is not tools used to build project,  but is runtime target which can be deployed to a device.

File: recipes/xlsxio/all/conanfile.py (hunk 3 above)
Comment (hiiizxf) [reply recipes/xlsxio/all/conanfile.py:41]: OK

**Review by danimtb — APPROVED** [review]

**Pull request merged**